test/integration/puppetlabs/services/jruby/tasks_test.clj

```diff
@@ -0,0 +1,115 @@
+(ns puppetlabs.services.jruby.tasks-test
+  (:require [clojure.test :refer :all]
+            [puppetlabs.kitchensink.core :as ks]
+            [puppetlabs.services.jruby.jruby-puppet-testutils :as jruby-testutils]
+            [puppetlabs.services.jruby.puppet-environments :as puppet-env]
+            [puppetlabs.services.master.master-core :as mc]
+            [me.raynes.fs :as fs]
+            [cheshire.core :as cheshire]
+            [schema.core :as schema]
+            [schema.test :as schema-test]
+            [puppetlabs.puppetserver.testutils :as testutils]
+            [puppetlabs.trapperkeeper.app :as tk-app]
+            [puppetlabs.trapperkeeper.testutils.bootstrap :as tk-bootstrap])
+  (:import (java.io File)))
+
+(use-fixtures :once schema-test/validate-schemas)
+
+(def TaskOptions
+  {:name schema/Str
+   :module-name schema/Str
+   :metadata? schema/Bool
+   :number-of-files schema/Num})
+
+(schema/defn gen-task
+  "Assumes tasks dir already exists -- generates a set of task files for a
+  single task."
+  [env-dir :- (schema/cond-pre File schema/Str)
+   task :- TaskOptions]
+  (let [task-name (:name task)
+        extensions (cycle '(".rb" "" ".sh" ".exe" ".py"))
+        task-dir (fs/file env-dir "modules" (:module-name task) "tasks")]
+    (fs/mkdirs task-dir)
+    (when (:metadata? task)
+      (fs/create (fs/file task-dir (str task-name ".json"))))
+    (dotimes [n (:number-of-files task)]
+      (fs/create (fs/file task-dir (str task-name (nth extensions n ".rb")))))))
```

Comment:
I interpreted `:number-of-files` as an upper-unbounded number of files to create but this will create the same five files repeatedly if passed a number > 5. Intentional?